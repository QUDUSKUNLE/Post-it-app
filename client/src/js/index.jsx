import React from 'react'
import ReactDOM from 'react-dom'
//import { browserHistory } from 'react-router'
// import { HashRouter as Router, browserHistory } from 'react-router-dom'
// import routes from './components/routes'
// import { Provider } from 'react-redux'
// import thunk from "redux-thunk";
// import { createStore, applyMiddleware } from 'redux';
// import { composeWithDevTools } from 'redux-devtools-extension';
// import rootReducer from '../src/reducers';
import App from './components/App'

// const store = createStore(
// 	rootReducer,
// 	composeWithDevTools(
// 		applyMiddleware(thunk)
// 	)
// );

ReactDOM.render( < App/>
	, document.getElementById('app')
);
