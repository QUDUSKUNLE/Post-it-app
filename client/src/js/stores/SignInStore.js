import { EventEmitter } from 'events';
import AppDispatcher from '../dispatcher/AppDispatcher.js';
import {
  SIGN_IN_SUCCESS, SIGN_IN_ERROR
} from '../constants/ActionConstants.js';

/**
 * Signout Store, it hold user's state, listen to signin Actions
 * @class SignInStore
 */
class SignInStore extends EventEmitter {
  constructor(props) {
    super(props);
    this.handleActions = this.handleActions.bind(this);
  }

  handleActions(action) {
    switch (action.type) {
      case SIGN_IN_SUCCESS:
        this.emit('SIGN_OUT_SUCCESS');
        break;

      case SIGN_IN_ERROR:
        this.emit('SIGN_IN_ERROR');
        break;

      default:
    }
  }
}

// Initiate an instance of SignInStore
const signInStore = new SignInStore();

// Register a dispatcher and bind it to handleActions method
AppDispatcher.register(signInStore.handleActions.bind(signInStore));

// export an instance of SignInStore
export default signInStore;
