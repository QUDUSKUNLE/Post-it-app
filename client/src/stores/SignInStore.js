import { EventEmitter } from 'events';
import AppDispatcher from '../dispatcher/AppDispatcher';
import {
  SIGN_IN_SUCCESS,
  GOOGLE_SIGN_IN_SUCCESS, PASSWORD_RESET_SUCCESS
} from '../constants/ActionConstants';

/**
 * Signout Store, it hold user's state, listen to signin Actions
 * @class SignInStore
 */
class SignInStore extends EventEmitter {
  /**
   * @constructor
   */
  constructor() {
    super();
    this.signInMessage = {};
    this.googleSignInMessage = {};
    this.passwordResetResponse = {};
    this.signInUser = this.signInUser.bind(this);
    this.googleSignIn = this.googleSignIn.bind(this);
    this.passwordReset = this.passwordReset.bind(this);
    this.handleActions = this.handleActions.bind(this);
  }

  signInUser() {
    return this.signInMessage;
  }

  googleSignIn() {
    return this.googleSignInMessage;
  }

  passwordReset() {
    return this.passwordResetResponse;
  }

  /**
   * Receives actions and update the stores accordingly
   * @method handleActions
   * @param {object} action - Action type and data
   * @return {*} -
   */
  handleActions(action) {
    switch (action.type) {
      case SIGN_IN_SUCCESS:
        this.signInMessage = action.response;
        this.emit('SIGN_IN_SUCCESS');
        break;

      case GOOGLE_SIGN_IN_SUCCESS:
        this.googleSignInMessage = action.response;
        this.emit('GOOGLE_SIGN_IN_SUCCESS');
        break;

      case PASSWORD_RESET_SUCCESS:
        this.passwordResetResponse = action.response;
        this.emit('PASSWORD_RESET_SUCCESS');
        break;
      default:
    }
  }
}

// Initiate an instance of SignInStore
const signInStore = new SignInStore();

// Register a dispatcher and bind it to handleActions method
AppDispatcher.register(signInStore.handleActions.bind(signInStore));

// export an instance of SignInStore
export default signInStore;
