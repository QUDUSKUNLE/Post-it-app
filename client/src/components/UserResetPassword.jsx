import React from 'react';
import toastr from 'toastr';
import resetPassword from '../actions/resetPasswordActions.js';
import SignInStore from '../stores/SignInStore.js';

/**
 * @description - renders ResetPassword Component
 * @class ResetPassword
 * @extends {React.Component}
 */
export default class UserResetPassword extends React.Component {
  /**
   * Create a constructor
   * @constructor
   * @param {object} props -
   */
  constructor(props) {
    super(props);
    this.state = { email: '' };

    /**
     * @description This binding is necessary to make `this` work
     * in the callback
     */
    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
    this.handlePasswordReset = this.handlePasswordReset.bind(this);
  }

  componentDidMount() {
    SignInStore.on('PASSWORD_RESET_SUCCESS', this.handlePasswordReset);
  }

  componentWillUnmount() {
    SignInStore.removeListener('PASSWORD_RESET_SUCCESS',
      this.handlePasswordReset);
  }

  /**
   * onChange event.
   * @param {object} event The first number.
   * @returns {void} bind input values to name.
   */
  onChange(event) {
    this.setState({
      [event.target.name]: event.target.value
    });
  }

  /**
	 * @description This handles resetPassword form submission
	 * @param {object} event .
	 * @returns {void} .
	 */
  onSubmit(event) {
    event.preventDefault();
    const resetEmail = { email: this.state.email };
    resetPassword(resetEmail);
  }

  /**
   * @description - This handles reset Password Action
   * @returns {Object} Object
   * @MemberOf Reset Password
   */
  handlePasswordReset() {
    const passwordResetResponse = SignInStore.passwordReset();
    toastr.success(passwordResetResponse.message);
  }

  /**
   * @description - render method, React lifecycle method
   * @returns {Object} ResetPassword component
   * @ResetPassword
   */
  render() {
    return (
      <div className="passwordreset">
        <h5 className="text-center">
          <b>Reset your password</b>
        </h5>
        <div className="container resetform">
          <div className="row">
            <div className="col-md-6 col-md-offset-3 w3-card w3-white">
              <form onSubmit={this.onSubmit} id="resetform">
                <div className="form-group">
                  <label htmlFor="email">
                    Enter email
                  </label>
                  <input value={this.state.email}
                    onChange={this.onChange}
                    id="email" type="email"
                    className="signinform" placeholder="johndoe@example.com"
                    name="email" required />
                </div>
                <button type="submit"
                  className="signinformbtn">Send
                </button>
              </form>
            </div>
          </div>
        </div>
      </div>
    );
  }
}
