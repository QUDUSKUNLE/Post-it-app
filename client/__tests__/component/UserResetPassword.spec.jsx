import React from 'react';
import expect from 'expect';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { mount, shallow } from 'enzyme';
import sinon from 'sinon';
import UserResetPassword
  from '../../src/components/UserResetPassword.jsx';
import '../../../server/config/index.js';
import SignInStore from '../../src/stores/SignInStore.js';

describe('<UserResetPassword/>', () => {
  let wrapper;
  beforeEach(() => {
    const res = { response: 'Password reset email sent successfully!' };
    SignInStore.passwordReset = jest.fn(() => res);

    const props = {
      email: '',
    };

    wrapper = mount(<UserResetPassword {...props}/>,
      {
        childContextTypes: { router: PropTypes.object },
        context: {
          router:
          {
            history: {
              push: () => null,
              replace: () => null,
              createHref: () => null,
              path: '/passwordreset',
              component: '[function UserResetPassword]',
              location: {
                pathname: '/passwordreset',
                search: '',
                hash: '',
                key: 'zo06fn'
              },
              computedMatch: {
                path: '/passwordreset',
                url: '/passwordreset',
                isExact: true,
                params: {}
              }
            }
          }
        }
      }
    );
  });
  it('component should have empty initial states', () => {
    expect(wrapper.state()).toBeDefined();
    expect(wrapper.state().email).toEqual('');
  });
  it('component should be defined', () => {
    expect(UserResetPassword).toBeDefined();
  });
  it('component should render correctly', () => {
    const component = shallow(<UserResetPassword/>);
    expect(component).toMatchSnapshot();
    expect(component.find('form')).toHaveLength(1);
    expect(component.find('input')).toHaveLength(1);
    expect(component.find('button')).toHaveLength(1);
    expect(component.find('input.signinform')).toHaveLength(1);
  });
  it('component should call onChange method', () => {
    const event = { target: { name: 'name', value: 'value' } };
    wrapper.instance().onChange(event);
    expect(wrapper.state().name).toEqual('value');
  });
  it('component should call onSubmit method', () => {
    wrapper.instance().onSubmit({ preventDefault() {} });
    expect(wrapper.state().email).toEqual('');
  });
  it('component should call componentDidMount component lifecycle', () => {
    const spy = sinon.spy(UserResetPassword.prototype, 'componentDidMount');
    wrapper.instance().componentDidMount();
    expect(spy.calledOnce).toBeTruthy();
  });
  it('component should call componentWillUnmount component lifecycle', () => {
    const spy = sinon.spy(UserResetPassword.prototype, 'componentWillUnmount');
    wrapper.instance().componentWillUnmount();
    expect(spy.calledOnce).toBeTruthy();
  });
  it('sets state when handlePasswordReset method is called', () => {
    expect(wrapper.node.state).toEqual({ email: '' });
    wrapper.instance().handlePasswordReset();
    expect(wrapper.nodes[0].onChange).toBeDefined();
    expect(wrapper.nodes[0].onSubmit).toBeDefined();
    expect(wrapper.nodes[0].handlePasswordReset).toBeDefined();
  });
  it('should redirect to another page on click of a button', () => {
    wrapper.find(Link).at(2).simulate('click');
  });
});
