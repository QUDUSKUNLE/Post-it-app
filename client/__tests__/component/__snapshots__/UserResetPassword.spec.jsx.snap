// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserResetPassword/> component should render correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="passwordreset"
  >
    <h5
      className="text-center"
    >
      <b>
        Reset your password
      </b>
    </h5>
    <div
      className="container resetform"
    >
      <div
        className="row"
      >
        <div
          className="col-md-6 col-md-offset-3 w3-card w3-white"
        >
          <form
            id="resetform"
            onSubmit={[Function]}
          >
            <div
              className="form-group"
            >
              <label
                htmlFor="email"
              >
                Enter email
              </label>
              <input
                className="signinform"
                id="email"
                name="email"
                onChange={[Function]}
                placeholder="johndoe@example.com"
                required={true}
                type="email"
                value=""
              />
            </div>
            <button
              className="signinformbtn"
              type="submit"
            >
              Send
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>,
  "nodes": Array [
    <div
      className="passwordreset"
    >
      <h5
        className="text-center"
      >
        <b>
          Reset your password
        </b>
      </h5>
      <div
        className="container resetform"
      >
        <div
          className="row"
        >
          <div
            className="col-md-6 col-md-offset-3 w3-card w3-white"
          >
            <form
              id="resetform"
              onSubmit={[Function]}
            >
              <div
                className="form-group"
              >
                <label
                  htmlFor="email"
                >
                  Enter email
                </label>
                <input
                  className="signinform"
                  id="email"
                  name="email"
                  onChange={[Function]}
                  placeholder="johndoe@example.com"
                  required={true}
                  type="email"
                  value=""
                />
              </div>
              <button
                className="signinformbtn"
                type="submit"
              >
                Send
              </button>
            </form>
          </div>
        </div>
      </div>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <UserResetPassword />,
      "_debugID": 40,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": UserResetPassword {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "handlePasswordReset": [Function],
        "onChange": [Function],
        "onSubmit": [Function],
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "email": "",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="passwordreset"
        >
          <h5
            className="text-center"
          >
            <b>
              Reset your password
            </b>
          </h5>
          <div
            className="container resetform"
          >
            <div
              className="row"
            >
              <div
                className="col-md-6 col-md-offset-3 w3-card w3-white"
              >
                <form
                  id="resetform"
                  onSubmit={[Function]}
                >
                  <div
                    className="form-group"
                  >
                    <label
                      htmlFor="email"
                    >
                      Enter email
                    </label>
                    <input
                      className="signinform"
                      id="email"
                      name="email"
                      onChange={[Function]}
                      placeholder="johndoe@example.com"
                      required={true}
                      type="email"
                      value=""
                    />
                  </div>
                  <button
                    className="signinformbtn"
                    type="submit"
                  >
                    Send
                  </button>
                </form>
              </div>
            </div>
          </div>
        </div>,
        "_debugID": 41,
        "_renderedOutput": <div
          className="passwordreset"
        >
          <h5
            className="text-center"
          >
            <b>
              Reset your password
            </b>
          </h5>
          <div
            className="container resetform"
          >
            <div
              className="row"
            >
              <div
                className="col-md-6 col-md-offset-3 w3-card w3-white"
              >
                <form
                  id="resetform"
                  onSubmit={[Function]}
                >
                  <div
                    className="form-group"
                  >
                    <label
                      htmlFor="email"
                    >
                      Enter email
                    </label>
                    <input
                      className="signinform"
                      id="email"
                      name="email"
                      onChange={[Function]}
                      placeholder="johndoe@example.com"
                      required={true}
                      type="email"
                      value=""
                    />
                  </div>
                  <button
                    className="signinformbtn"
                    type="submit"
                  >
                    Send
                  </button>
                </form>
              </div>
            </div>
          </div>
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <UserResetPassword />,
}
`;
